### PR TITLE
Polymorphic filter select

### PR DESCRIPTION
**Pull request by shekibobo**
This was adapted from #2461 and updates it to the latest master, along with a few other fixes.

Polymorphic associations are still `belongs_to` associations, so they should behave as much the same way as possible. Now, when we update the type field of a polymorphic association, we get a autocomplete field for the association record instead of it loading the entire collection (which causes browser lockup for extremely large collections).

### CODE REVIEW

**Comment by shekibobo** [comment]
I'm pretty sure the remaining test failure isn't related to this PR. If you rebuild the failing one, it will most likely pass.

**Comment by shekibobo** [comment]
Just merged with master, so this is up to date and passing now.

**Comment by arnvald** [comment]
@shekibobo thank you for taking this code and updating it with newest master 🙇  I was going to update my fork and your work saved me a lot of time!

@mshibuya can anything be done to help getting this merged to master?
I've been using this code in production for a long time and it's been working fine. I haven't finished adding support for `associated_collection_cache_all` or `associated_collection_scope`, but I feel it's still an improvement over current version which stops working for big databases.

**Comment by grillermo** [comment]
Thank you for this, i'm using it on my fork

File: app/views/rails_admin/main/_form_polymorphic_association.html.haml [hunk 1]
```diff
@@ -2,10 +2,25 @@
...
-  collection = field.associated_collection(selected_type)
   selected = field.bindings[:object].send(field.association.name)
+  collection = selected ? [[field.formatted_value, selected.id]] : [[]]
   column_type_dom_id = form.dom_id(field).sub(field.method_name.to_s, type_column)
+  current_action = params[:action].in?(['create', 'new']) ? 'create' : 'update'
+
+  default_options = { float_left: false }
+
+  js_data = type_collection.inject({}) do |options, model|
+    model_name = model[0].downcase.parameterize.underscore
```
[the remark this answers is not in the document]
Comment (shekibobo) [reply app/views/rails_admin/main/_form_polymorphic_association.html.haml]: Why `.second`?

**Comment by robwise** [comment]
What steps need to be taken to get this merged?

**Comment by mshibuya** [comment]
Fantastic, thanks!

**Pull request merged**